ligo/src/treasuryDAO.mligo

```diff
@@ -0,0 +1,121 @@
+// SPDX-FileCopyrightText: 2020 TQ Tezos
+// SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+#include "defaults.mligo"
+#include "types.mligo"
+#include "proposal.mligo"
+#include "helper/unpack.mligo"
+#include "base_DAO.mligo"
+
+#include "treasuryDAO/types.mligo"
+
+// -------------------------------------
+// Helpers
+// -------------------------------------
+
+let unpack_transfer_type_list (key_name, p : string * ((string, bytes) map)) : transfer_type list =
+  let b = unpack_key (key_name, p)
+  in match ((Bytes.unpack b) : ((transfer_type list) option)) with
+    Some (v) -> v
+  | None -> (failwith "UNPACKING_NOT_TRANSFER_TYPE_LIST" : transfer_type list)
+
+// -------------------------------------
+// Configuration Lambdas
+// -------------------------------------
+
+let treasury_DAO_proposal_check (params, store : propose_params * storage) : bool =
+  let proposal_size = Bytes.size(Bytes.pack(params.proposal_metadata)) in
+  let frozen_scale_value = unpack_nat("a", store.extra) in
+  let frozen_extra_value = unpack_nat("b", store.extra) in
+  let max_proposal_size = unpack_nat("s_max", store.extra) in
+  let min_xtz_amount = unpack_tez("x", store.extra) in
```

Comment:
Shouldn't `min_xtz_amount` be unpacked from `y` key?

Reply:
Fixed